### ISSUE TITLE
False positive warning about "too many arguments for format string"

### ISSUE THREAD

**Issue by deivid-rodriguez** [oracle/truffleruby#3938]
In CRuby:

```
$ ruby -vwe 'sprintf("%*d", 20, 14)'
ruby 3.4.5 (2025-07-16 revision 20cda200d3) +PRISM [arm64-darwin24]
```

In Truffleruby:

```
$ ruby -vwe 'sprintf("%*d", 20, 14)'
truffleruby 24.2.1, like ruby 3.3.7, Oracle GraalVM Native [arm64-darwin20]
-e:1: warning: too many arguments for format string
```

**Comment by fniephaus** [comment]
Thanks for filing the ticket, @deivid-rodriguez!

**Comment by deivid-rodriguez** [comment]
This patch seems to do the trick for me, in case it helps!

```diff
diff --git a/src/main/java/org/truffleruby/core/kernel/KernelNodes.java b/src/main/java/org/truffleruby/core/kernel/KernelNodes.java
index 7a41743175..413fc44053 100644
--- a/src/main/java/org/truffleruby/core/kernel/KernelNodes.java
+++ b/src/main/java/org/truffleruby/core/kernel/KernelNodes.java
@@ -1710,6 +1710,9 @@ public abstract class KernelNodes {
                     for (var config : configs) {
                         if (!config.isLiteral()) {
                             modifiersCount++;
+                            if (config.isWidthStar() || config.isPrecisionStar()) {
+                              modifiersCount++;
+                            }
                         }
                     }
 
```

**Comment by fniephaus** [comment]
Thanks for the patch!

**Comment by eregon** [comment]
TruffleRuby 33 [has been released](https://truffleruby.dev/blog/truffleruby-33-is-released) with this fix.
